common/djangoapps/xblock_django/user_service.py

```diff
@@ -55,6 +57,28 @@ def get_anonymous_user_id(self, username, course_id):
         course_id = CourseKey.from_string(course_id)
         return anonymous_id_for_user(user=user, course_id=course_id, save=False)
 
+    def get_enrollments(self, course_id, **kwargs):
+        """
+        Return iterator of enrollments, as dicts.
+        """
+        enrollments = CourseEnrollment.objects.filter(
+            course_id=course_id).select_related('programcourseenrollment')
+        for enrollment in enrollments:
+            enrd = {
+                'user_id': enrollment.user.id,
+                'username': enrollment.user.username,
+                'full_name': enrollment.user.profile.name,
+                'email': enrollment.user.email,
+                'enrolled': enrollment.is_active,
+                'track': enrollment.mode,
+            }
+            try:
+                pce = enrollment.programcourseenrollment.program_enrollment
+                enrd['student_uid'] = pce.external_user_key
```

Comment:
I am curious, why not check pce for null and possibly use a ternary operator instead of a potential exception?

Reply:
both `enrollment.programcourseenrollment` and `programcourseenrollment.program_enrollment` may raise an exception. A ternary operator wouldn't cut it here. I would have to check `programcourseenrollment_id` and `program_enrollment_id` for null, which gets a bit messy.